### PR TITLE
[Merged by Bors] - feat: port `Traversable` deriving handlers

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #5737

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Komyyy** [comment]
Now, it works.
All I have to do is have to refactoring.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5737
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5737~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Tactic/DeriveTraversable.lean [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+    return f'.app e
+  else if ←
+      (match t with
+        | .app t' _ => withNewMCtxDepth <| isDefEq t' cl
+        | _ => return false) then
+    mkAppM ``Comp.mk #[e]
+  else
+    return e
+
+def getAuxDefOfDeclName : TermElabM FVarId := do
```
Comment (semorrison) [inline Mathlib/Tactic/DeriveTraversable.lean:52]: Unfortunately the docBlame linter was temporarily disabled during the port. Nevertheless, could you make sure that #lint is happy?

**Comment by kmill** [comment]
Looks good! I made some small changes and added a little documentation while reading through to keep my focus -- that's a lot of Lean 4 core I'm not familiar with, but it seems sensible. Thanks for undertaking this long and hard battle!

bors r+

**Pull request closed without merging**